### DISCUSSION TITLE
How does React handle prop drilling, and what are the downsides of excessive prop drilling?

### DISCUSSION THREAD

**Discussion by Rashid-Aly** [vercel/next.js#69049] [App Router] [answered]
How does React handle prop drilling, and what are the downsides of excessive prop drilling?

**Comment by kashif-hussain6** [comment, marked as answer]
Prop drilling is when you pass data from a parent component to deeply nested child components through props. While React lets you do this, it can make your code harder to maintain and understand. Too much prop drilling can make it tough to follow the data flow and can cause extra re-renders. To avoid these problems, you can use the Context API or state management tools like Redux.
